File: explorer/src/schema/types/company.ts
import {GraphQLList, GraphQLObjectType, GraphQLString} from 'graphql';
import {UserType} from './user';
import axios from "axios";

export const CompanyType: GraphQLObjectType = new GraphQLObjectType({
    name: 'Company',
    fields: () => ({
        id: {type: GraphQLString},
        name: {type: GraphQLString},
        description: {type: GraphQLString},
        users: {
            type: new GraphQLList(UserType as typeof UserType),  // Explicitly annotate the type here
            async resolve(parentValue, args) {
                let res = await axios
                    .get(`http://localhost:3001/companies/${parentValue.id}/users`);
                return await res.data;
            }
        }
    }),
});

File: explorerj/src/main/java/io/klabs/explorerj/model/Company.java
package io.klabs.explorerj.model;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
public class Company {
    private String id;
    private String name;
    private String description;
    private List<User> users;
}
